frontend/src/metabase/admin/settings/components/widgets/AdminSettingInput.tsx

```diff
@@ -151,10 +152,25 @@ export function BasicAdminSettingInput({
       );
     case "radio":
       return (
-        <Radio.Group id={name} value={localValue} onChange={handleChange}>
-          {options?.map(({ label, value }) => (
-            <Radio key={value} value={value} label={label} />
-          ))}
+        <Radio.Group
+          id={name}
+          value={String(localValue)}
+          onChange={(newValue) => {
+            if (options && hasBooleanOptions(options)) {
```

Comment:
@iethree I implemented this slightly differently than we discussed.

I figured it would be extra-safe to check that there were only two options, one "true" and one "false" value, before auto-converting